go/cmd/dolt/commands/profile.go

```diff
@@ -223,6 +235,24 @@ func prettyPrintProfile(profileName string, profile Profile) {
 		profileName, profile.User, profile.Password, profile.Host, profile.Port, profile.NoTLS, profile.DataDir, profile.DoltCfgDir, profile.PrivilegeFile, profile.BranchControl, profile.UseDB))
 }
 
+// setGlobalConfigPermissions sets permissions on global config file to 0600 to protect potentially sensitive information (credentials)
+func setGlobalConfigPermissions(dEnv *env.DoltEnv) error {
+	homeDir, err := env.GetCurrentUserHomeDir()
+	if err != nil {
+		return errhand.BuildDError("error: failed to get home directory: %s", err).Build()
+	}
+	path, err := dEnv.FS.Abs(filepath.Join(homeDir, dbfactory.DoltDir, env.GlobalConfigFile))
+	if err != nil {
+		return errhand.BuildDError("error: failed to get global config path: %s", err).Build()
+	}
+	err = os.Chmod(path, 0600)
```

Comment:
Did you verify that this works? Not clear why the sql-server.lock file permissions are wrong even though we do the same thing there